[FSDP][Easy] Move utils to `_utils.py`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84212

I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

Commit:
[FSDP][Easy] Move utils to `_utils.py`

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/_utils.py:
```diff
@@ -78,3 +83,51 @@ def f(module: torch.nn.Module, prefix: str, *args, **kwargs):
+
+
+@torch.no_grad()
+def _alloc_storage(tensor: torch.Tensor, size: torch.Size) -> bool:
```
I originally used the `bool` return value, but I no longer do. I kept them for now since they might be useful in the future, but I can remove them if you want.

rohan-varma commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -4164,35 +4168,6 @@ def _get_default_cuda_device(module: nn.Module) -> torch.device:
-    ), "Then tensor storage should have been resized to be 0."
-    data.storage().resize_(size.numel())  # type: ignore[attr-defined]
-
-def p_assert(cond: Any, s: Any) -> None:
```
hm, I thought I added an argument to this function in one of my recent PRs, but not sure if this is rebased

awgu replied:
The added argument is there now after rebasing.

rohan-varma reviewed (approved):
LGTM

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot rebase -s

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot merge -g

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot rebase -s

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot rebase -s

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot rebase -s

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot merge -g

Commit:
Update on "[FSDP][Easy] Move utils to `_utils.py`"


I pulled this out into a separate PR. This just moves some utility functions to `_utils.py`.

[ghstack-poisoned]

awgu commented:
@pytorchbot merge -f The failing MacOS build tests appear to be flaky. I have tried rebasing onto `viable/strict` several times, but these build tests continue to fail when checking out the repo. These should be unrelated to this PR.

awgu commented:
@pytorchbot merge -f "The failing MacOS build tests appear to be flaky. I have tried rebasing onto viable/strict several times, but these build tests continue to fail when checking out the repo. These should be unrelated to this PR."

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -102,7 +106,7 @@
     "FullyShardedDataParallel", "ShardingStrategy", "MixedPrecision",
     "CPUOffload", "BackwardPrefetch", "StateDictType", "StateDictConfig",
     "FullStateDictConfig", "LocalStateDictConfig", "ShardedStateDictConfig",
-    "OptimStateKeyType", "TrainingState_", "p_assert", "clean_tensor_name",
```
Removing this to avoid the public bindings from failing.
